features/steps/steps.py

```diff
@@ -171,6 +171,33 @@ def when_i_fix_a_issue_by_enabling_service(context, issue):
     )
 
 
+@when("I fix `{issue}` by updating expired token")
+def when_i_fix_a_issue_by_updating_expired_token(context, issue):
+    token = getattr(context.config, "contract_token")
+    when_i_run_command(
+        context=context,
+        command="ua fix {}".format(issue),
+        user_spec="with sudo",
+        stdin="r\n{}\n".format(token),
+    )
+
+
+@when("I update contract to use `{contract_field}` as `{new_value}`")
+def when_i_update_contract_field_to_new_value(
+    context, contract_field, new_value
+):
+    when_i_run_command(
+        context,
+        'sed -i \'s/"{}": "[^"]*"/"{}": "{}"/g\' {}'.format(
+            contract_field,
+            contract_field,
+            new_value,
+            DEFAULT_MACHINE_TOKEN_PATH,
+        ),
```

Comment:
Fun! My first thought would've been to read in the file, parse it as json into a dict, make the update, and then serialize it back to json. But this works just as well for any string value. I manually double checked it would handle commas correctly and it did.

Will this ever need to be used for a non-string json value?

Reply:
I don't think it should, since the sed command here would not work well for other types. I think the best way for us is to add another function that handles that if needed

Reply:
Yep that would work. Then we would probably rename this to something like `when_i_update_contract_field_to_new_str_value` and the nonstring version would become something like `when_i_update_contract_field_to_new_nonstr_value`. But I don't think we need to do that preemptively.

Reply:
We could even keep the function as is and just redirect in that method based on the type of `new_value`. But I agree that we must implement this when needed